<!--
Εξαίρεση είναι ένα αντικείμενο που περιγράφει ένα σφάλμα ή μια απροσδόκητη συμπεριφορά ενός σεναρίου PHP.

Οι εξαιρέσεις δημιουργούνται από πολλές λειτουργίες και κλάσεις PHP.

Οι καθορισμένες από τον χρήστη λειτουργίες και κλάσεις μπορούν επίσης να ρίξουν εξαιρέσεις.

Οι εξαιρέσεις είναι ένας καλός τρόπος για να σταματήσετε μια συνάρτηση όταν συναντά 
δεδομένα που δεν μπορεί να χρησιμοποιήσει.
-->

<!--
try {
    κώδικα που μπορεί να ρίξει (throw) εξαιρέσεις
} catch(Exception $e) {
    κώδικα που εκτελείται όταν εντοπίζεται μια εξαίρεση
} finally {
    κώδικα που τρέχει πάντα ανεξάρτητα από το εάν έχει εξαχθεί μια εξαίρεση
}
-->

<!--
Η δήλωση throw επιτρέπει σε μια καθορισμένη από τον χρήστη συνάρτηση ή μέθοδο να ρίξει μια εξαίρεση.
Όταν μια εξαίρεση ρίχνεται, ο κώδικας που ακολουθεί δεν θα εκτελεστεί.

Εάν δεν εξαχθεί μια εξαίρεση, θα προκύψει ανεπανόρθωτο σφάλμα με το μήνυμα "Uncaught Exception".
-->

<!DOCTYPE html>
<html>
<body>

<?php
function divide($dividend, $divisor) {
  if($divisor == 0) {
    throw new Exception("Division by zero");
  }
  return $dividend / $divisor;
}

try {
  echo divide(5, 0);
} catch(Exception $e) {
  echo "Unable to divide. ";
} finally {
  echo "Process complete.";
}
?>

</body>
</html>
